### ISSUE TITLE
Fix Products by Category Block showing different results when a single category is selected depending on the 'Display products matching' attribute

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce#42676]
## To reproduce

1. Add a Products by Category block.
2. Select a category (_Clothing_, for example).
3. Press Done and notice several products appear.
4. Now, edit the block (pencil icon).
5. Select another category (_Accessories_, for example).
6. Set the option to 'Display products matching: All selected categories'.
7. Unselect the second category and click on Done.
8. Notice now there are fewer products than before.

## Expected behavior
If only one category is selected and, because of that, the `Display products matching` option is hidden, that option should have no effect on what products are shown.

## Screenshots
![Peek 2020-08-31 10-51](https://user-images.githubusercontent.com/3616980/91701750-06265600-eb78-11ea-8993-f7300dc014fb.gif)

**Comment by WunderBart** [comment]
Not sure if this is still relevant, @Aljullu. FWIW, I cannot repro with the Product Collectioin block filters. Do you think we can close this one?

**Comment by Aljullu** [comment]
Yes, we can close this issue as the _Products by Category_ block is deprecated. Thanks for catching this issue, @WunderBart!
